monkestation/code/modules/reagents/chemistry/machinery/atmos_machine.dm

```diff
@@ -0,0 +1,172 @@
+#define REAGENTS_BASE_VOLUME 100 // actual volume is REAGENTS_BASE_VOLUME plus REAGENTS_BASE_VOLUME * rating for each matterbin
+
+///dont tell anyone this but this is literally just smoke machine code but with names replaced and smoke replaced with gas -Borbop
+
+/obj/machinery/atmos_machine
+	name = "evaporation machine"
+	desc = "A machine with a centrifuge installed into it. It produces gas with any reagents you put into the machine."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "smoke0"
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/atmos_machine
+
+
+
+	var/efficiency = 10
+	var/on = FALSE
+	var/cooldown = 0
+	var/useramount = 30 // Last used amount
+	var/setting = 1 // displayed range is 3 * setting
```

Comment:
The setting setup was designed more for the smoke machine, not the atmos machine.
Doesn't make much sense to say "12 tiles" when it's not creating smoke.

Reply:
yea i use the smoke machine menu currently because i didn't wanna make  a new tgui, can fix this pretty easily though